lib/openstudio-standards/standards/ashrae_90_1_prm/ashrae_90_1_prm.Model.rb

```diff
@@ -590,4 +590,54 @@ def thermal_zone_prm_lab_delta_t(thermal_zone)
   def model_get_fan_power_breakdown
     return true
   end
+
+  # Template method for adding a setpoint manager for a coil control logic to a heating coil.
+  # ASHRAE 90.1-2019 Appendix G.
+  #
+  # @param model [OpenStudio::Model::Model] Openstudio model
+  # @param thermalZones Array([OpenStudio::Model::ThermalZone]) thermal zone array
+  # @param coil Heating Coils
+  # @return [Boolean] true
+  def model_set_central_preheat_coil_spm(model, thermalZones, coil)
+    # search for the highest zone setpoint temperature
+    max_heat_setpoint = 0.0
+    coil_name = coil.name.get.to_s
+    thermalZones.each do |zone|
+      tstat = zone.thermostatSetpointDualSetpoint
+      if tstat.is_initialized
+        tstat = tstat.get
+        setpoint_sch = tstat.heatingSetpointTemperatureSchedule
+        setpoint_min_max = search_min_max_value_from_design_day_schedule(setpoint_sch, 'heating')
+        setpoint_c = setpoint_min_max['max']
+        if setpoint_c > max_heat_setpoint
+          max_heat_setpoint = setpoint_c
+        end
+      end
+    end
+
+    max_heat_setpoint_f = OpenStudio.convert(max_heat_setpoint, 'C', 'F').get
+    preheat_setpoint_f = max_heat_setpoint_f - 20
+    preheat_setpoint_c = OpenStudio.convert(preheat_setpoint_f, 'F', 'C').get
+    # in this situation, we hard set the temperature to be 52 F or 11.1 C
+    # (ASHRAE 90.1 Room heating stepoint temperature is 72 F)
+    preheat_setpoint_c = 11.1 if preheat_setpoint_c == 0.0
```

Comment:
What is this line doing? Why would preheat_setpoint_c be zero?

Reply:
Ah, thanks for catching this line.

The line 603 sets a default value to max_heat_setpoint = 0.0,
so the line 623 originally was coded to prevent failure from searching the max design setpoint. But I forgot to update the condition after adding the unit conversions.

In this case, I can move this code to line [617](https://github.com/NREL/openstudio-standards/blob/383a1e050ecdfbb350815b850bb8e51ef16d8f37/lib/openstudio-standards/standards/ashrae_90_1_prm/ashrae_90_1_prm.Model.rb#L617):
max_heat_setpoint= 72 if max_heat_setpoint== 0.0 (assuming no heating setpoint temperature is lower than 0 degree C.
and the rest will be the same.

Reply:
Sounds good!